django/utils/deprecation.py

```diff
@@ -83,6 +85,181 @@ def __new__(cls, name, bases, attrs):
         return new_class
 
 
+def deprecate_posargs(category, *, moved):
```

Comment:
Reviewer note: please leave feedback on `@deprecate_posargs` in #19145, not this PR.